### ISSUE TITLE
List item in markdown with checkbox is not correctly indented

### ISSUE THREAD

**Issue by isodude** [mattermost/mattermost#22550]
#### Summary
List item in markdown with checkbox is not correctly indented

#### Steps to reproduce
In chat, type
```
- list-item
- [ ] list-item
- list-item
```

#### Expected behavior
- list-item
- [x] list-item
- list-item

Note here that the dot is replaced by the checkbox.

#### Observed behavior (that appears unintentional) 
- list-item
  :white_square_button: list-item
- list-item

Note here that the list item is indented as a list item but the checkbox is included in the list-item and does not replace the dot in itself.

#### Note
It seems that https://github.com/mattermost/mattermost-server/commit/355056622b3bd1cd6e1f3dda18273a473bc14823 fixes this last time around.
My fix this to make it like how github do it is to add
```css
li.list-item--task-list {
  margin-left: -15px;
}
```
As I go through that I see that even github misaligns the list!
Since the dot is much smaller than the checkbox, the checkbox will push the list-time text to the right.
```css
ul.markdown__list li.not(list-item--task-list) span {
  margin-left: 2px;
}
```
Fixes this somewhat.

- - [ ] list-item

ends up with a linebreak between the dot and the checkbox.
If I remove the span inside
```
<ul class="markdown__list">
::marker
<span><ul class="markdown__list"><ul></span>
</ul>
```

It behaves better.

**Comment by isodude** [comment]
Regarding task list in a markdown__list I assume it needs some fixing inside https://github.com/mattermost/mattermost-server/blob/355056622b3bd1cd6e1f3dda18273a473bc14823/web/react/utils/markdown.jsx#L226

**Comment by amyblais** [comment]
Thanks, opened https://mattermost.atlassian.net/browse/MM-51488.

**Comment by hchorfispiria** [comment]
@amyblais Would you mind assigning this issue to me?

**Comment by hchorfispiria** [comment]
After investigating and debugging the issue in code and database, everything seems good, there is no issue.
The message is correctly saved in the database.
<img width="264" alt="Mattermost_List_With_Tasks_DB" src="https://user-images.githubusercontent.com/65722778/234038838-388a82d4-7ead-47ca-92d6-a631117da807.png">

According to the official documentation on Format Messages (URL below)
https://docs.mattermost.com/channels/format-messages.html
Making a task list by including square brackets [ ] has the following format:
```text
- [ ] Item one
- [ ] Item two
- [x] Completed item
```
Result:
- [ ] Item one
- [ ] Item two
- [x] Completed item

That means creating a single task (with a checkbox) has the following format:
```text
- [ ] Item one
```
So, the "-" is part of the syntax. We cannot create a checkbox without "-"

Similarly, we can create a list using the following syntax:
```text
- Item one
- Item two
- Completed item
```
Result:
- Item one
- Item two
- Completed item

In conclusion, this is not a bug, and to create a list of tasks (with checkboxes) we should use the following syntax with a double "-":
```text
- list-item
- - [ ] list-item
- list-item
```

Result:
- list-item
- - [ ] list-item
- list-item

On Mattermost.
<img width="600" alt="Mattermost_List_With_Tasks_Syntax" src="https://user-images.githubusercontent.com/65722778/234039032-4703dbea-77de-4390-9482-4dc2ba575dcd.png">
<img width="600" alt="Mattermost_List_With_Tasks_Result" src="https://user-images.githubusercontent.com/65722778/234039059-b5629d6c-5a2f-45fd-804e-95c4207b216a.png">

**Comment by isodude** [comment]
I don't understand your reasoning here. This is not a database issue, it's a css margin problem.

Why should a list not be able to contain a checkbox?

It's perfectly supported here on github, aligned ok.

- item 1
- [ ] item 2
- [x] item 3
- item 4

**Comment by hmhealey** [comment]
I agree that this is a CSS issue and not something else, so I'll open this back up.

Looking back at that previous commit which attempted to fix this, I can't figure out why it's using sibling selectors for the CSS or using a `:before` to make a fake list marker in some cases. This seems like it should just be a matter of modifying the margins on list items so that the real marker on regular list items matches the size of the positioning of the bullet point

**Comment by hchorfispiria** [comment]
Sure, there is a possible fix with CSS. However, since the marked CSS is generated by the marked library itself, we should address it to marked.
The fix with CSS is to modify the `list-item--task-list` class or remove it completely, but this may affect the formatting of other lists.
It is not good practice to play directly with marked CSS. It's up to marked to generate CSS. Or at least using best practice like this: https://github.com/markedjs/marked/issues/119

**Comment by hmhealey** [comment]
They may not actually have this issue in more recent versions of marked. We're on a heavily modified version of marked (https://github.com/mattermost/marked) because, at some point, marked was inactive and I ended up forking it and making a ton of changes for features we needed at the time.

Because we've diverged so far from them, I think we're fine to keep modifying the CSS ourselves. We could also add our own classes to the output if we want to change those instead, but I don't think that's strictly necessary either

**Comment by drashtiv03** [comment]
Can you please assign me this issue.I would love to work on it. @amyblais

**Comment by isodude** [comment]
For some reason this is now worse.
This is now the result of
```
- list-item
- [ ] list-item
- list-item
```
- list-item
  :white_square_button: list-item
&nbsp;• list-item

**Comment by drashtiv03** [comment]
I have unassigned myself due to some other prior commitments.Sorry for the inconvenience.

**Comment by cinlloc** [comment]
I would like to work on this.
